chainio/clients/builder.go

```diff
@@ -51,11 +51,100 @@ type Clients struct {
 	PrometheusRegistry          *prometheus.Registry  // Used if avs teams need to register avs-specific metrics
 }
 
-func BuildAll(
+// This is a struct that holds all the params that are needed to interact
+// with the AVS and EL contracts without the Wallet/Private key.
+// This is useful for read-only operations, such as fetching metrics.
+type ClientsBase struct {
+	AvsRegistryChainReader      *avsregistry.ChainReader
+	AvsRegistryChainSubscriber  *avsregistry.ChainSubscriber
+	AvsRegistryChainWriter      *avsregistry.ChainWriter
+	ElChainReader               *elcontracts.ChainReader
+	EthHttpClient               eth.HttpBackend
+	EthWsClient                 eth.WsBackend
+	AvsRegistryContractBindings *avsregistry.ContractBindings
+	EigenlayerContractBindings  *elcontracts.ContractBindings
+	Metrics                     *metrics.EigenMetrics // exposes main avs node spec metrics that need to be incremented by avs code and used to start the metrics server
+	PrometheusRegistry          *prometheus.Registry
+}
+
+// This is a struct that extends ClientsBase with Wallet and TxManager.
+// This is useful for the full set of operations (reads and writes), such as sending transactions.
+type ClientsWithPkKey struct {
```

Comment:
Why is this added as a new struct? Is the old `Clients` still used? If not, let's get rid of Clients, and just rename this new struct `Clients`.